coffee_github/src/utils.rs

```diff
@@ -25,7 +25,7 @@ pub async fn clone_recursive_fix(repo: git2::Repository, url: &URL) -> Result<()
     Ok(())
 }
 
-pub async fn git_upgrade(
+pub async fn git_upgrade_with_branch(
```

Comment:
why changing the name here?

Reply:
because I added another function that uses the git commit to upgrade

Reply:
I do not see the point of changing the name, due that there is no git upgrade command in `git`

Reply:
Moreover, git uses always a branch there is never a `without_branch` cases